nameko/containers.py

```diff
@@ -134,8 +144,16 @@ def __init__(self, service_cls, config, worker_ctx_cls=None):
         self.service_cls = service_cls
         self.config = config
 
-        if worker_ctx_cls is None:
-            worker_ctx_cls = WorkerContext
+        if worker_ctx_cls is not None:
+            warnings.warn(
```

Comment:
Definitely agree with this. Something like two major releases or 6 months, whichever comes last?

I'll create document somewhere to describe which deprecation warnings we have and when they were introduced, so we know when they're eligible for removal.